src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -131,33 +147,146 @@ internal static bool TryExtractBaggage(string[] baggageCollection, out Dictionar
 
                 if (pair.IndexOf('=') < 0)
                 {
-                    continue;
+                    baggage = null;
+                    return false;
                 }
 
                 var parts = pair.Split(EqualSignSeparator, 2);
                 if (parts.Length != 2)
```

Comment:
List-member could contain multiple `=` signs. For example, 
1. in property.

> list-member            =  key OWS "=" OWS value *( OWS ";" OWS property )
property               =  key OWS "=" OWS value
property               =/ key OWS

2. [A value can also contain equal signs.](https://github.com/open-telemetry/opentelemetry-go/blob/59bbe11641bb8af130ecf0a4bb48a2b11479285c/baggage/baggage.go#L285C23-L285C48)

Reply:
Ad 1. Properties are not handled/recognized properly currently, this is something that needs to be addressed, but as it requires changes in both Baggage API and propagator, I'd prefer to address it in a separate PR.
Ad 2. This will be handled properly - see test case added in https://github.com/open-telemetry/opentelemetry-dotnet/pull/5647/commits/6b88989ce3243593c4db0beea6774f039ea4222e

Reply:
I created [issue](https://github.com/open-telemetry/opentelemetry-dotnet/issues/5677) to track handling of properties.
Currently, properties attached to a baggage entry are treated as a part of a value.
With baggage entry value validation added in this PR (`ValidateValue`), such entries will be considered invalid due to the existence of `;` char. This might need to be adjusted.